logstash-core/lib/logstash/pipeline.rb

```diff
@@ -447,12 +447,14 @@ def worker_loop(batch_size, batch_delay)
       shutdown_requested |= signal.shutdown? # latch on shutdown signal
 
       batch = @filter_queue_client.read_batch # metrics are started in read_batch
```

Comment:
@guyboertje I agree during filtering the batch could become effectively empty, but it seems strange that `read_batch` would ever return an empty batch. Can you remind me why we wanted this?